### DISCUSSION TITLE
[Bug]: V7 Incorrect typing error, when passing control as a prop

### DISCUSSION THREAD

**Discussion by Steelrover** [react-hook-form/react-hook-form#6436] [Q&A]
### Version Number

7.14.2

### Codesandbox/Expo snack

https://codesandbox.io/s/xenodochial-khorana-vnh3c?file=/src/App.tsx

### Steps to reproduce

1. Use custom type `FormValues` with no optional fields for `useForm`
2. Try to pass `control `as a prop to underlying form component, which expects a prop of type `Сontrol`
3. Typescript will report that `Type 'Control<FormValues, object>' is not assignable to type 'Control<Record<string, any>, object>'`

### Expected behaviour

No typescript errors for control prop.

This error isn't present on v6.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
you are missing the generic 

```tsx
import React, { FC } from "react";
import { Controller, Control } from "react-hook-form";

type ComponentProps = {
  control: Control<{
    firstName: string;
    secondName: string;
  }>;
};

const Component: FC<ComponentProps> = ({ control }) => {
  return (
    <Controller
      control={control}
      name="secondName"
      render={({ field: { onChange, value } }) => {
        return (
          <input value={value} onChange={onChange} placeholder="Second Name" />
        );
      }}
    />
  );
};

export default Component;
```

**Reply by Steelrover** [reply]
But what if I need to create a universal component (like a custom version of select for example) that is not bound to a single form?

And this error doesn't appear if all fields in type are optional

**Reply by bluebill1049** [reply]
You will have to pass down the generic, we are using template string literal to check against input names.
